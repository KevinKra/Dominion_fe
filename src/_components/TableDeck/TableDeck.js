import React, { Component } from 'react';
import TreasureCards from '../TreasureCards/TreasureCards';
import VictoryCards from '../VictoryCards/VictoryCards';
import ActionCards from '../ActionCards/ActionCards';
import Notification from '../Notification/Notification';
import { connect } from 'react-redux';
import * as actions from '../../_redux/actions/index';
import './TableDeck.scss';

export class TableDeck extends Component {
  componentDidMount() {
    this.props.updateTableCards(this.props.tableCards);
  }

  buyCard = e => {
    if (this.props.buyingPower < 1) return;
    const clickedName = e.target.closest('article').getAttribute('data-name');
    const matchingStack = this.props.tableCards.find(tableCard => {
      return tableCard.name === clickedName;
    });
    if (this.props.spendingPower < matchingStack.cost) {
      return console.log('not enough money to buy');
    } else {
      const boughtId = matchingStack.id_list.pop();
      this.props.boughtCard(clickedName);
      this.props.spendTreasure(matchingStack.cost);
      this.props.bought(boughtId);
      this.props.useBuy();
    }
  };

  render() {
    return (
      this.props.tableCards.length !== 0 && (
        <section className="TableDeck">
          <Notification />
          <VictoryCards buyCard={this.buyCard} />
          <ActionCards buyCard={this.buyCard} />
          <TreasureCards buyCard={this.buyCard} />
        </section>
      )
    );
  }
}

const mapStateToProps = state => ({
  tableCards: state.tableCards,
  spendingPower: state.spendingPower,
  buyingPower: state.buyingPower
});

const mapDispatchToProps = dispatch => ({
  updateTableCards: cards => dispatch(actions.updateTableCards(cards)),
  spendTreasure: value => dispatch(actions.spendTreasure(value)),
  bought: id => dispatch(actions.bought(id)),
  boughtCard: cardName => dispatch(actions.boughtCard(cardName)),
  useBuy: () => dispatch(actions.useBuy())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TableDeck);
