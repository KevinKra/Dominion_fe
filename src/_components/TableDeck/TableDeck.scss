@import "../../styles/_variables.scss";
@import "../../styles/_animations.scss";

.TableDeck {
  display: flex;
  position: relative;
  z-index: 3;
  width: 100%;
  height: 100%;
  background: url("https://tsdbcg.herokuapp.com/card_images/freshwater-bay.jpg");
  background-size: cover;
  border-bottom: 4px solid $color2;
  animation: fadeInDown 0.8s;
}
