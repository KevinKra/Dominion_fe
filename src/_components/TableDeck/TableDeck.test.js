import React from "react";
import { TableDeck } from "./TableDeck";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("<TableDeck />", () => {
  let wrapper;
  const mockUpdateTableCards = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <TableDeck updateTableCards={mockUpdateTableCards} tableCards={1} />
    );
  });
  it("should match the snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
