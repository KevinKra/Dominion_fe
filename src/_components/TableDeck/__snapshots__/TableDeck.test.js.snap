// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableDeck /> should match the snapshot 1`] = `
<section
  className="TableDeck"
>
  <Connect(Notification) />
  <Connect(VictoryCards)
    buyCard={[Function]}
  />
  <Connect(ActionCards)
    buyCard={[Function]}
  />
  <Connect(TreasureCards)
    buyCard={[Function]}
  />
</section>
`;
